nsot/models/resource.py

```diff
@@ -69,40 +77,73 @@ def set_query(self, query, site_id=None, unique=False):
                 log.debug('Regex enabled for %r' % name)
 
             # Attribute lookup params
-            params = dict(
-                name=name, resource_name=resource_name
-            )
+            if name not in concrete_field_names:
+                params = dict(
+                    name=name, resource_name=resource_name
+                )
+            else:
+                params = dict()
+                params[name] = value
             # Only include site_id if it's set
             if site_id is not None:
                 params['site_id'] = site_id
 
             # If an Attribute doesn't exist, the set query is invalid. Return
             # an empty queryset. (fix #99)
-            try:
-                attr = Attribute.objects.get(
-                    **params
-                )
-            except Attribute.DoesNotExist as err:
-                raise exc.ValidationError({
-                    'query': '%s: %r' % (err.message.rstrip('.'), name)
-                })
+            if name in concrete_field_names:
```

Comment:
This block is harder to follow because in the previous and following `if..else` you're evaluating `if name not in concrete_field_names...else` (you're evaluating attribute fields first), but in this one you're evaluating `if name in concrete_field_names` (evaluating concrete fields first). 

Could you please adjust  them to consistently evaluate in the same order, whichever you decide is best? 

Or, if there's an explicit reason you chose to evaluate concrete fields first in this block, could you please add some commenting to explain why?

Reply:
Done. Now we're always evaluating Attributes first.